docs/src/tutorials/conic/optimal-power-flow.jl

```diff
@@ -0,0 +1,286 @@
+# Copyright 2017, Iain Dunning, Joey Huchette, Miles Lubin, and contributors    #src
+# This Source Code Form is subject to the terms of the Mozilla Public License   #src
+# v.2.0. If a copy of the MPL was not distributed with this file, You can       #src
+# obtain one at https://mozilla.org/MPL/2.0/.                                   #src
+
+# # Optimal power flow
+
+# **This tutorial was originally contributed by James Foster (@jd-foster).**
+
+# This tutorial formulates and solves an optimal power flow problem,
+# a much-studied nonlinear problem from the field of electrical engineering.
+# Our particular focus is on obtaining a good estimate of the objective value
+# at the global optimum through the use of semidefinite programming techniques.
+
+# The tutorial highlights JuMP's ability to formulate problems involving
+# complex-valued decision variables and complex matrix cones such as the 
+# [`HermitianPSDCone`](@ref) object.
+# See the [Complex number support](@ref) section of the manual for more details.
+
+# For another example of modeling with complex decision variables
+# see the [Quantum state discrimination](@ref) tutorial.
+
+# ## Required packages
+
+# This tutorial requires the following packages:
+
+using JuMP
+import Ipopt
+import SCS
+import LinearAlgebra: diag, diagm
+import SparseArrays: sparse, spdiagm, sparsevec
+import DataFrames: DataFrame, rename
+import Test
+
+# ## Formulation
+
+# Optimal power flow problems for electrical transmission typically pose the
+# following question: what is the most cost-effective operation of electricity
+# generators while meeting constraints on the safe limits of network components?
+
+# We will use the 9 _bus_ network test case `case9mod` to explore this problem
+# following the example of Krasko and Rebennack (2017).
+
+# Here *bus* and network *node* are taken as analogous terms, as are *branch* and transmission *line*.
+
+# The graph of the network shows three nodes each used for the different purposes of
+# generation ``G`` (nodes 1, 2 and 3),
+# trans-shipment (nodes 4, 6 and 8) 
+# and demand ``L`` (5, 7, and 9).
+
+# ![Nine Nodes](../../assets/case9mod.png)
+
+# For future reference, let's name the number of nodes in the network:
+N = 9
+
+# The network data can be summarised as follows.
+
+# Generation power bounds (active and reactive):
+P_G_lb = sparsevec([1, 2, 3], [10, 10, 10], N)
+P_G_ub = sparsevec([1, 2, 3], [250, 300, 270], N)
+
+Q_G_lb = sparsevec([1, 2, 3], [-5, -5, -5], N)
+Q_G_ub = sparsevec([1, 2, 3], [300, 300, 300], N)
+
+# Power demand levels (active, reactive and complex form):
+P_D_fx = sparsevec([5, 7, 9], [54, 60, 75], N)
+Q_D_fx = sparsevec([5, 7, 9], [18, 21, 30], N)
+
+S_D_fx = P_D_fx + im * Q_D_fx
```

Comment:
This has always been my hesitation to add such a tutorial. These acronyms are pretty meaningless to the casual reader.

Reply:
It is a bit cryptic. I'll change `_D_fx` -> `Demand`